Makefile

```diff
@@ -42,40 +42,55 @@
 #
 #
 ###############################################################################
-# The build architecture is select by setting the ARCH variable.
-# For example: When building on ppc64le you could use ARCH=ppc64le make <....>.
-# When ARCH is undefined it defaults to amd64.
-ARCH?=amd64
-ifeq ($(ARCH),amd64)
-	ARCHTAG?=
-	GO_BUILD_VER?=v0.9
-	PROTOC_VER?=v0.1
-	FV_TYPHAIMAGE?=calico/typha:v0.6.0-beta1-16-g512a0f2
-endif
+# Both native and cross architecture builds are supported.
+# The target architecture is select by setting the ARCH variable.
+# When ARCH is undefined it is set to the detected host architecture.
+# When ARCH differs from the host architecture a crossbuild will be performed.
 
-ifeq ($(ARCH),ppc64le)
-	ARCHTAG:=-ppc64le
-	GO_BUILD_VER?=latest
-	PROTOC_VER?=latest
-	FV_TYPHAIMAGE?=calico/typha-ppc64le:latest
-endif
 
-ifeq ($(ARCH),s390x)
-	ARCHTAG:=-s390x
-	GO_BUILD_VER?=latest
-	FV_TYPHAIMAGE?=calico/typha-s390x:latest
+# BUILDARCH is the host architecture
+# ARCH is the target architecture
+# we need to keep track of them separately
+BUILDARCH ?= $(shell uname -m)
+
+# canonicalized names for host architecture
+ifeq ($(BUILDARCH),aarch64)
+        BUILDARCH=arm64
+endif
+ifeq ($(BUILDARCH),x86_64)
+        BUILDARCH=amd64
 endif
 
-GO_BUILD_CONTAINER?=calico/go-build$(ARCHTAG):$(GO_BUILD_VER)
-PROTOC_CONTAINER?=calico/protoc$(ARCHTAG):$(PROTOC_VER)
+# unless otherwise set, I am building for my own architecture, i.e. not cross-compiling
+ARCH ?= $(BUILDARCH)
 
-# Name used to tag the Felix container image when building.
-FELIX_IMAGE_NAME?=calico/felix$(ARCHTAG):latest
+# canonicalized names for target architecture
+ifeq ($(ARCH),aarch64)
+        override ARCH=arm64
+endif
+ifeq ($(ARCH),x86_64)
+    override ARCH=amd64
+endif
 
-# Name of the images to run FV tests against.
-FV_FELIXIMAGE?=$(FELIX_IMAGE_NAME)
-FV_ETCDIMAGE?=quay.io/coreos/etcd:v3.2.5$(ARCHTAG)
-FV_K8SIMAGE?=gcr.io/google_containers/hyperkube$(ARCHTAG):v1.7.5
+GO_BUILD_VER ?= latest
```

Comment:
We need to get a pinned version in here.  I believe `v0.10` is the current `latest`

Reply:
It is `v0.10` (I just checked), and that is a good idea. Fixing.